app/controllers/submission_logs_controller.rb

```diff
@@ -17,7 +17,9 @@
 
 class SubmissionLogsController < ApplicationController
   before_action { |controller|
-    if params[:assignment_id]
+    if params[:assignment_id] and params[:submission_similarity_id]
+      controller.send :authenticate_custom_actions_for_teaching_staff, only: [:view_similarity]                                                                    
```

Comment:
Do you have specific documentation for what constitutes "teaching_staff" somewhere in your roles documentation?  (this seems like a very detailed controller function, so I want to know where all of the roles and documentation are).